### ISSUE TITLE
Question (OpenIdConnect): On Successful SignUp / Register event?

### ISSUE THREAD

**Issue by TheReaLee** [aspnet/AspNetCore#10390]
Is it possible to subscribe to an event when a user has successfully signed up/registered?

`

            options.Events = new OpenIdConnectEvents()
            {   
                OnRedirectToIdentityProvider = OnRedirectToIdentityProvider,
                OnRemoteFailure = OnRemoteFailure,
                OnAuthorizationCodeReceived = OnAuthorizationCodeReceived,
                OnAuthenticationFailed = OnAuthenticationFailed,
                OnMessageReceived = OnMessageReceived,
                OnRedirectToIdentityProviderForSignOut = OnRedirectToIdentityProviderForSignOut,
                OnRemoteSignOut = OnRemoteSignOut,
                OnSignedOutCallbackRedirect = OnSignedOutCallbackRedirect,
                OnTicketReceived = OnTicketReceived,
                OnTokenResponseReceived = OnTokenResponseReceived,
                OnTokenValidated = OnTokenValidated,
                OnUserInformationReceived = OnUserInformationReceived
                //OnSignUp/OnRegistered? or maybe a property from one of the events above which uniquely identifies a signup procedure?
            };

`

Basically our goal is to run some extra code when a user has signed up... ex validation etc...

**Comment by blowdart** [comment]
That is not something OIDC exposes, presumably for privacy reasons, because it shouldn't be your apps concern if a user is new to the identity provider or not,, so it's not something we can support.

**Comment by TheReaLee** [comment]
@blowdart FYI managed to get this working. Posted answer on stack overflow:
https://stackoverflow.com/questions/36097669/sign-up-event-from-azure-b2c/56250604#56250604
